packages/salesforce-adapter/src/filters/remove_memory_only_annotations.ts

```diff
@@ -0,0 +1,46 @@
+/*
+*                      Copyright 2020 Salto Labs Ltd.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with
+* the License.  You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+import {
+  Element, isObjectType,
+} from '@salto-io/adapter-api'
+import _ from 'lodash'
+import * as constants from '../constants'
+import { FilterCreator } from '../filter'
+
+const removeMemoryOnlyAnnotations = (elements: Element[]): void => {
+  elements
+    .flatMap(e => (isObjectType(e) ? [e, ...Object.values(e.fields)] : [e]))
```

Comment:
this is only relevant for object types and fields, we should filter and operate only on them, either by accepting `types: ObjectType[]` as a parameter to this function or by adding the filter here
```suggestion
  elements
    .filter(isObjectType)
    .flatMap(e => [e, ...Object.values(e.fields)])
```

Reply:
that's true for our specific annotation - do we want to bake this assumption into the filter?

Reply:
the annotation is salesforce specific and therefore cannot be set on instances (not part of `INSTANCE_ANNOTATIONS`). so as long as we don't plan to filter things out based on core annotations (I think that is a fairly safe assumption to make) we should be able to assume that we only care about annotations on fields and types